devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,198 @@
+
+### Getting help to resolve test failures
+
+Remember to state which tests are failing and for each of those
+* The detailed error message (taken from the TestNG tab)
+
+* Any relevant screenshots (e.g. screen shot of the diff window showing the offending difference)
+
+### Other errors and solutions
+
+1. **ERROR**: Tests fail randomly during dev server testing.
+
+  **SOLUTION**: Make sure there is only one dev server running. Eclipse will happily allow you to start multiple dev servers.
+
+  ![troubleshooting-test-4.png](images/troubleshooting-test-4.png)
+
+2. **ERROR (on Windows)**: NullPointerException when trying to access email box. The error message looks something like this:
+   ```
+      java.lang.NullPointerException at
+      javax.mail.internet.ParameterList.set(ParameterList.java:165) at
+      com.sun.mail.imap.protocol.BODYSTRUCTURE.parseParameters(BODYSTRUCTURE.java:390)
+   ```
+  **SOLUTION**: In eclipse, move the javax.mail.....jar (from Oracle) to the top of the build class path.
+
+   (Properties → Java build path → 'order and export' tab)
+
+3. **ERROR**: Test failure message encountered when running full test suite: "Selenium cannot find Firefox binary in PATH".
+
+  **SOLUTION 1**: Path to Firefox executable on local machine is incorrect. Specify the correct folder in system PATH variable.
+  
+   → Open Windows Explorer → Right-click on Computer → Advanced System Settings → "Advanced" tab → Environment Variables… → Select "PATH" from the list → Add directory of "Mozilla Firefox" folder to "Variable value" field
+
+  **SOLUTION 2**: Incorrect custom path in test.firefox.path. Make sure that the path is set correctly following the example from test.template.properties
+
+4. **ERROR**: After deploying on the staging server, you get an error related to … Unsupported major.minor version ... when trying to access pages of the staged app.
+This can happen if you have Java 8 is installed and Eclipse uses Java 8 during deployment, even if you have already configured the project to use Java 7.
+Refer [this page](http://java.wildstartech.com/Java-Platform-Standard-Edition/mac-os-x-java-development/how-to-configure-eclipse-to-run-with-java-7-when-java-8-is-installed) to learn how to modify eclipse.ini to use Java 7 by default.
+
+5. **ERROR**: A handful of failed test cases (<10).
+
+  **SOLUTION**: Ensure compatible version of Firefox is installed as specified under Prerequisites.
+6. **ERROR**: Browser alert about invalid ssl certificate.
+
+  **SOLUTION**: Ensure that test.app.url in your test.properties uses -dot- instead of '.' when using secondary subdomains.
+  Eg. http\://4-30-dot-teammates-john.appspot.com
```

Comment:
This link becomes live, which we don't really want. Wrap it in back-quotes to make it code-like, i.e `http://4-30-dot-teammates-john.appspot.com`